pymaster/field.py

```diff
@@ -694,21 +694,32 @@ def __init__(self, positions, weights, field, spin=None, beam=None,
         self.positions = np.array(positions, dtype=np.float64)
         self.weights = np.array(weights, dtype=np.float64)
         if np.shape(self.positions) != (2, len(self.weights)):
-            raise ValueError("Weights must be 2D array of shape"
+            raise ValueError("Positions must be 2D array of shape"
                              " (2, len(weights)).")
-        if not (np.logical_and(self.positions[0] > 0.,
+        if not lonlat and not (np.logical_and(self.positions[0] > 0.,
                                self.positions[0] < np.pi)).all():
             raise ValueError("First dimension of positions must be colatitude"
                              " in radians, between 0 and pi.")
-        if not (np.logical_and(self.positions[1] > 0.,
+        if not lonlat and not (np.logical_and(self.positions[1] > 0.,
                                self.positions[1] < 2.*np.pi)).all():
             raise ValueError("Second dimension of positions must be longitude"
                              " in radians, between 0 and 2*pi.")
+        if lonlat and not (np.logical_and(self.positions[0] > 0.,
+                                          self.positions[0] < 360.)).all():
```

Comment:
We probably need to allow some angles to be `==0` or `==360`, in case they happen. Same above for 0 and 2pi. And same for theta/dec